### PR TITLE
bpf: fix hw_csum issue for icmp probe packets

### LINKED ISSUE #9482: cilium_host: hw csum failure
Reported via slack: https://cilium.slack.com/archives/C1MATJ5U5/p1571823875167100

```
and the OS -- Ubuntu 18.04.2 LTS   kernel: 5.3.0-1004-azure

I'm running this on self hosted cluster in azure 

This is what the configuration looks like
helm template cilium \
    --namespace kube-system \
    --set global.nodePort.enabled=true \
    --set global.hostServices.enabled=true \
    --set global.tunnel=vxlan \
    --set global.tag=v1.6.3 \
> cilium.yaml
```

```
[  601.889533] cilium_host: hw csum failure
[  601.894231] skb len=36 headroom=128 headlen=36 tailroom=28
               mac=(114,14) net=(128,20) trans=148
               shinfo(txflags=0 nr_frags=0 gso(size=0 type=0 segs=0))
               csum(0x0 ip_summed=2 complete_sw=0 valid=1 level=0)
               hash(0x5e1df61c sw=0 l4=1) proto=0x0800 pkttype=0 iif=7
[  601.926422] dev name=cilium_host feat=0x0x000030884fdd59e9
[  601.935407] skb linear:   00000000: 45 00 00 24 b3 1a 00 00 3f 01 a5 fc 0b 10 04 ed
[  601.941761] skb linear:   00000010: 0b 10 07 b6 00 00 0a 00 78 73 fc 73 15 d0 23 08
[  601.948612] skb linear:   00000020: 41 5a 06 e6
[  601.952386] CPU: 0 PID: 1673 Comm: oneagentnetwork Not tainted 5.3.0-1004-azure #4-Ubuntu
[  601.952387] Hardware name: Microsoft Corporation Virtual Machine/Virtual Machine, BIOS 090007  06/02/2017
[  601.952388] Call Trace:
[  601.952389]  <IRQ>
[  601.952394]  dump_stack+0x4d/0x6a
[  601.952397]  netdev_rx_csum_fault.part.0+0x41/0x45
[  601.952399]  netdev_rx_csum_fault.cold+0xb/0x10
[  601.952402]  __skb_checksum_complete+0xd9/0xf0
[  601.952405]  ? skb_send_sock_locked+0x270/0x270
[  601.952408]  ? reqsk_fastopen_remove+0x150/0x150
[  601.952412]  nf_ip_checksum+0xe4/0x110
[  601.952422]  nf_conntrack_icmpv4_error+0x13d/0x150 [nf_conntrack]
[  601.952425]  ? kfree_skbmem+0x4e/0x60
[  601.952428]  ? kfree_skb+0x3a/0xa0
[  601.952436]  nf_conntrack_in.cold+0x1d/0x83 [nf_conntrack]
[  601.952440]  ? tcf_classify+0x40/0x100
[  601.952446]  ipv4_conntrack_in+0x14/0x20 [nf_conntrack]
[  601.952449]  nf_hook_slow+0x45/0xb0
[  601.952453]  ip_rcv+0x90/0xd0
[  601.952456]  ? ip_rcv_finish_core.isra.0+0x3c0/0x3c0
[  601.952459]  __netif_receive_skb_one_core+0x87/0xa0
[  601.952461]  __netif_receive_skb+0x18/0x60
[  601.952463]  process_backlog+0x95/0x140
[  601.952467]  net_rx_action+0x12e/0x340
[  601.952470]  ? rcu_core+0x104/0x450
[  601.952473]  __do_softirq+0xdb/0x2ca
[  601.952477]  do_softirq_own_stack+0x2a/0x40
[  601.952478]  </IRQ>
[  601.952481]  do_softirq.part.0+0x30/0x40
[  601.952483]  __local_bh_enable_ip+0x50/0x60
[  601.952485]  _raw_spin_unlock_bh+0x1e/0x20
[  601.952489]  packet_getsockopt+0x255/0x410
[  601.952492]  ? apparmor_socket_getsockopt+0x29/0x50
[  601.952496]  __sys_getsockopt+0x8d/0x120
[  601.952499]  __x64_sys_getsockopt+0x25/0x30
[  601.952502]  do_syscall_64+0x68/0x1d0
[  601.952503]  ? prepare_exit_to_usermode+0x51/0xb0
[  601.952506]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[  601.952508] RIP: 0033:0x7efc4ada09aa
[  601.952510] Code: 48 8b 0d e1 84 2c 00 f7 d8 64 89 01 48 83 c8 ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 49 89 ca b8 37 00 00 00 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d ae 84 2c 00 f7 d8 64 89 01 48
[  601.952512] RSP: 002b:00007efc4856a568 EFLAGS: 00000246 ORIG_RAX: 0000000000000037
[  601.952514] RAX: ffffffffffffffda RBX: 000056428d4a9d20 RCX: 00007efc4ada09aa
[  601.952515] RDX: 0000000000000006 RSI: 0000000000000107 RDI: 0000000000000009
[  601.952516] RBP: 000056428d4a9ab0 R08: 00007efc4856a574 R09: 000056428d4a9d20
[  601.952517] R10: 00007efc4856a578 R11: 0000000000000246 R12: 000056428cb73b3c
[  601.952518] R13: 0000000000000053 R14: 00007efc4aa1be90 R15: 0000000000000053
```

### PR DESCRIPTION
**Pull request by borkmann**
Fixes: #9482

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: bpf/lib/nat.h [hunk 1]
```diff
@@ -356,6 +356,7 @@ static __always_inline int snat_v4_rewrite_egress(struct __ctx_buff *ctx,
 					return DROP_WRITE_ERROR;
 				from = tuple->sport;
 				to = state->to_sport;
+				flags = 0;
```
Comment (brb) [inline bpf/lib/nat.h]: A small comment why we set `flags` to zero would be useful for future readers.

**Review by brb — APPROVED** [review]
Good find! Just one minor request for a code comment :rocket:

**Review by pchaigno — APPROVED** [review]

**Comment by borkmann** [comment]
(travis CI has an outage, hence stuck at yellow, rest green)

**Review by kkourt — APPROVED** [review]

**Pull request merged**